autobahn/twisted/component.py

```diff
@@ -323,104 +317,23 @@ def _connect_transport(self, reactor, transport, session_factory):
         transport_endpoint = _create_transport_endpoint(reactor, transport.endpoint)
         return transport_endpoint.connect(transport_factory)
 
-    # XXX think: is it okay to use inlineCallbacks (in this
-    # twisted-only file) even though we're using txaio?
-    @inlineCallbacks
-    def start(self, reactor=None):
+    def start(self, reactor=None, log_level=None):
         """
```

Comment:
remove `log_level=` kwarg

Reply:
This was a workaround for point 3:
https://github.com/crossbario/autobahn-python/pull/914#issuecomment-349085777 but could be removed if the proposed changes to txaio are accepted.

Reply:
Hmmmmmmmm. That might take some more thinking. Because: allowing all `.debug()` calls to call all the way through to Python's logger if you've not got debug turned on will be (I believe) a fairly significant performance impact.

(Looking at this another way: if you *do* want debug messages to get through, why can't you call ``txaio.start_logging('debug')` or similar?)

The real point of *not* having `log_level=` kwarg in `start()` is because the rationale is: if you're calling `start()` yourself, you're "doing the logging stuff" (and managing the loop yourself). If you don't want to care about logging and loop-management, you'd use the high-level `run()` method of starting your components.

Also, consider the case where you have more than one `Component` -- what's supposed to happen if you pass different `log_level=` kwargs to each `start()` method? (With the code as it exists in this branch, whichever one got called most-recently will set the txaio global log level -- which I think is surprising and wrong).